### DISCUSSION TITLE
WinUI 3 vs WinUI 2

### DISCUSSION THREAD

**Discussion by dimohy** [microsoft/WindowsAppSDK#3144] [Feedback wanted]
Still, WinUI 3 is much slower than WinUI 2. I have a hard time accepting this.

Since WinUI 3 runs on win32, it's a bit of a loss because to use existing WinRT code.

But even so, it's too slow. Can you feel the speed difference between WinUI 3 and WinUI 2?

(After installing both WinUI 2 Gallery and WinUI 3 Gallery from the store, you can check the performance on the same page.)

**Comment by megabytesme** [comment]
I know it has nearly been a year, however it is still missing most things which make development on Windows good... I was just trying to use WinUI 3 however even something like using the designer isn't possible in VS2022. I thought that was it however the apps are literally slower than WinUI 2.8 apps... Completely ridiculous IMO, I'll be sticking with UWP (and hope something as good is phased in for when MS drops support for it). Fingers crossed WinUI 3 comes to UWP (unless that bridge has completely been crossed now)

**Reply by DarranRowe** [reply]
Out of curiosity, are you comparing C++/WinRT to C++/WinRT or C# to C#.
At the moment, the latter comparison is problematic. If you don't know why, I would suggest you look into .NET Native AOT.

**Reply by megabytesme** [reply]
Ah, interesting. I'm comparing C# to C#... I'll definitely read that! Thanks
